### DISCUSSION TITLE
v1.88.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#5200] [Announcements]
# v1.88.0

> [!warning]
> **BREAKING CHANGES**
> `immich-proxy` and `immich-web` are no longer used as [announced](https://github.com/immich-app/immich/discussions/5086). Please see the content that needs to be edited from the `docker-compose.yml` file below. `immich-server` now serves the api on `/api` and the web-app from `/`.
>
> The steps to update are as follow:
> 1. Bring down the stack with `docker compose down --remove-orphans`
> 2. Update the `docker-compose.yml` file
> 2.1. Remove `immich-proxy` service
> 2.2. Remove `immich-web` service
> 2.3. Expose port `2283:3001` in the `immich-server` service
> 3. Run `docker compose pull`
> 4. Bring up the stack with `docker compose up -d`
>
> For those using a **custom** proxy, please update the routing to forward all requests to `immich-server` without the `/api` path re-write.

```diff
services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
+   ports:
+     - 2283:3001
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-microservices:
     [...]
  immich-machine-learning:
     [...]

-  immich-web:
-    container_name: immich_web
-    image: ghcr.io/immich-app/immich-web:${IMMICH_VERSION:-release}
-    env_file:
-      - .env
-    restart: always

  typesense:
    [...]

  redis:
    [...]

  database:
    [...]

- immich-proxy:
-   container_name: immich_proxy
-   image: ghcr.io/ Warninimmich-app/immich-proxy:${IMMICH_VERSION:-release}
-   ports:
-     - 2283:8080
-   depends_on:
-     - immich-server
-     - immich-web
-   restart: always

```

## Highlights

Welcome to release `v1.88.0` of Immich. In this release we have simplified how Immich is deployed, beautified our app interface with a _new font_, namely [Overpass](https://overpassfont.org/), and finally released _version 2_ of our [command line interface (CLI)](https://immich.app/docs/features/command-line-interface), which has been in the works for the past six months. 

<p align="center">
<img src="https://github.com/immich-app/immich/assets/27055614/3bb6d525-4768-4150-ba04-b2622d72fbf2" width="425" title="mobile-new-font"> 
<img src="https://github.com/immich-app/immich/assets/27055614/2074a08a-dd16-4942-afb8-31e6e3abf5ce" width="425" title="mobile-new-font"> 
<img src="https://github.com/immich-app/immich/assets/27055614/31761f17-a82d-4f46-b378-bdace55b4dbe" width="500" title="mobile-new-font"> 
</p>

<img width="1305" alt="image" src="https://github.com/immich-app/immich/assets/27055614/ba54b5f1-b0cf-4ac6-809a-5af957db6d2b">

<img width="1307" alt="image" src="https://github.com/immich-app/immich/assets/27055614/3d37537c-caa8-45ca-bafb-82ddd2f502a9">

- - - -

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Server
* feat(server): bmp format by @jrasm91 in https://github.com/immich-app/immich/pull/5197
* chore(server): update new CLI into the image by @alextran1502 in https://github.com/immich-app/immich/pull/5192

### CLI
* feat(cli): refactor and add album support by @etnoy in https://github.com/immich-app/immich/pull/4434
* docs(cli): Link to main Immich documentation by @etnoy in https://github.com/immich-app/immich/pull/5163
* chore(cli): push to npm by @jrasm91 in https://github.com/immich-app/immich/pull/5168
* chore(docs): Redirect old CLI paths by @bo0tzz in https://github.com/immich-app/immich/pull/5183
* chore(cli): set cli workdir in npm publish by @etnoy in https://github.com/immich-app/immich/pull/5185

### Web
* feat(web)!: SPA by @jrasm91 in https://github.com/immich-app/immich/pull/5069
* fix(web): improve year label position by @Funk66 in https://github.com/immich-app/immich/pull/5141
* fix(web): album sorting options by @martabal in https://github.com/immich-app/immich/pull/5127
* feat(web): new fonts by @alextran1502 in https://github.com/immich-app/immich/pull/5165
* chore(web): small font size improvement by @alextran1502 in https://github.com/immich-app/immich/pull/5190
* chore(web): remove deprecation message by @alextran1502 in https://github.com/immich-app/immich/pull/5115
* chore(web): album thumbnail size by @alextran1502 in https://github.com/immich-app/immich/pull/5196

### Mobile
* fix(mobile): Mark more strings for translation by @adamantike in https://github.com/immich-app/immich/pull/5132
* fix(mobile): use proper context for popping out from share by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5138
* chore(mobile): add initial DCM analysis_options by @shenlong-tanwen in https://github.com/immich-app/immich/pull/5136
* chore(mobile): Mobile make over by @alextran1502 in https://github.com/immich-app/immich/pull/5129
* refactor(mobile): riverpod codegen + riverpod lint by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4836

### Machine Learning
* chore(ml): update onnxruntime by @mertalev in https://github.com/immich-app/immich/pull/5175
* chore(ml): increase spool threshold by @mertalev in https://github.com/immich-app/immich/pull/5176

### Documentation and others
* docs: add warning to compose file and readme by @danieldietzler in https://github.com/immich-app/immich/pull/5123
* docs: Update machine-learning.md by @digitaljamie in https://github.com/immich-app/immich/pull/5160
* chore: fix typo in docker-compose.yml by @danieldietzler in https://github.com/immich-app/immich/pull/5124
* readme: Add Korean README by @bunubbv in https://github.com/immich-app/immich/pull/5128
* docs: update requirement for CLI by @alextran1502 in https://github.com/immich-app/immich/pull/5198

### Dependencies
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/5057
* chore(build): renovate grouping by @jrasm91 in https://github.com/immich-app/immich/pull/5167
* chore(deps): update redis:6.2-alpine docker digest to 80cc851 by @renovate in https://github.com/immich-app/immich/pull/5131
* chore(deps): update dependency @types/node to v20.9.2 by @renovate in https://github.com/immich-app/immich/pull/5139
* chore(deps): update dependency eslint to v8.54.0 by @renovate in https://github.com/immich-app/immich/pull/5140
* chore(deps): update docker/build-push-action action to v5.1.0 by @renovate in https://github.com/immich-app/immich/pull/5145
* chore(deps): update dependency @types/archiver to v6 by @renovate in https://github.com/immich-app/immich/pull/5146
* chore(deps): update dependency eslint-config-prettier to v9 by @renovate in https://github.com/immich-app/immich/pull/5173

## New Contributors
* @Funk66 made their first contribution in https://github.com/immich-app/immich/pull/5141
* @digitaljamie made their first contribution in https://github.com/immich-app/immich/pull/5160
* @bunubbv made their first contribution in https://github.com/immich-app/immich/pull/5128

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.87.0...v1.88.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.88.0'>v1.88.0</a>.</em>

**Comment by Uberpainer** [comment]
After upgrading to 1.88.1 is  the loading time for a album/shared album really long. Before it took like a split of a second to open the page. Not it take average of 28 seconds.

![image](https://github.com/immich-app/immich/assets/52506491/eb1d1977-2477-4c92-9815-a50a5d24b97c)
![image](https://github.com/immich-app/immich/assets/52506491/ce01a822-0618-42b9-8f44-b5b8efd0d7f0)

Error only happens on excising albums, if I create a new album does the issue not apear.

And this error from mobile app:
2023-11-21 10:49:46.260465,LogLevel.SEVERE,"ImmichErrorLogger","Catch all error: ApiException 400: HTTP connection failed: GET /album/09061d0f-fb67-490f-95dc-997d2829ff35 (Inner exception: Bad file descriptor)

#0      IOClient.send (package:http/src/io_client.dart:90)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      AlbumApi.getAlbumInfo (package:openapi/api/album_api.dart:311)
<asynchronous suspension>
#4      AlbumService.refreshRemoteAlbums.<anonymous closure> (package:immich_mobile/modules/album/services/album.service.dart:164)
<asynchronous suspension>
#5      SyncService._addAlbumFromServer (package:immich_mobile/shared/services/sync.service.dart:392)
<asynchronous suspension>
#6      diffSortedLists (package:immich_mobile/utils/diff.dart:30)
<asynchronous suspension>
#7      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/shared/services/sync.service.dart:267)
<asynchronous suspension>
 - ApiException 400: HTTP connection failed: GET /album/09061d0f-fb67-490f-95dc-997d2829ff35 (Inner exception: Bad file descriptor)

#0      IOClient.send (package:http/src/io_client.dart:90)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      AlbumApi.getAlbumInfo (package:openapi/api/album_api.dart:311)
<asynchronous suspension>
#4      AlbumService.refreshRemoteAlbums.<anonymous closure> (package:immich_mobile/modules/album/services/album.service.dart:164)
<asynchronous suspension>
#5      SyncService._addAlbumFromServer (package:immich_mobile/shared/services/sync.service.dart:392)
<asynchronous suspension>
#6      diffSortedLists (package:immich_mobile/utils/diff.dart:30)
<asynchronous suspension>
#7      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/shared/services/sync.service.dart:267)
<asynchronous suspension>
,#0      ApiClient.invokeAPI (package:openapi/api_client.dart:125)
<asynchronous suspension>
#1      AlbumApi.getAlbumInfo (package:openapi/api/album_api.dart:311)
<asynchronous suspension>
#2      AlbumService.refreshRemoteAlbums.<anonymous closure> (package:immich_mobile/modules/album/services/album.service.dart:164)
<asynchronous suspension>
#3      SyncService._addAlbumFromServer (package:immich_mobile/shared/services/sync.service.dart:392)
<asynchronous suspension>
#4      diffSortedLists (package:immich_mobile/utils/diff.dart:30)
<asynchronous suspension>
#5      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/shared/services/sync.service.dart:267)
<asynchronous suspension>
"
2023-11-21 10:49:46.260248,LogLevel.INFO,AlbumService"",""Syncing completed. Changes: false"","""""
2023-11-21 10:39:16.037308,LogLevel.SEVERE,"ImmichErrorLogger","Catch all error: ApiException 400: HTTP connection failed: GET /album/09061d0f-fb67-490f-95dc-997d2829ff35 (Inner exception: Bad file descriptor)

#0      IOClient.send (package:http/src/io_client.dart:90)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      AlbumApi.getAlbumInfo (package:openapi/api/album_api.dart:311)
<asynchronous suspension>
#4      AlbumService.refreshRemoteAlbums.<anonymous closure> (package:immich_mobile/modules/album/services/album.service.dart:164)
<asynchronous suspension>
#5      SyncService._addAlbumFromServer (package:immich_mobile/shared/services/sync.service.dart:392)
<asynchronous suspension>
#6      diffSortedLists (package:immich_mobile/utils/diff.dart:30)
<asynchronous suspension>
#7      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/shared/services/sync.service.dart:267)
<asynchronous suspension>
 - ApiException 400: HTTP connection failed: GET /album/09061d0f-fb67-490f-95dc-997d2829ff35 (Inner exception: Bad file descriptor)

#0      IOClient.send (package:http/src/io_client.dart:90)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      AlbumApi.getAlbumInfo (package:openapi/api/album_api.dart:311)
<asynchronous suspension>
#4      AlbumService.refreshRemoteAlbums.<anonymous closure> (package:immich_mobile/modules/album/services/album.service.dart:164)
<asynchronous suspension>
#5      SyncService._addAlbumFromServer (package:immich_mobile/shared/services/sync.service.dart:392)
<asynchronous suspension>
#6      diffSortedLists (package:immich_mobile/utils/diff.dart:30)
<asynchronous suspension>
#7      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/shared/services/sync.service.dart:267)
<asynchronous suspension>
,#0      ApiClient.invokeAPI (package:openapi/api_client.dart:125)
<asynchronous suspension>
#1      AlbumApi.getAlbumInfo (package:openapi/api/album_api.dart:311)
<asynchronous suspension>
#2      AlbumService.refreshRemoteAlbums.<anonymous closure> (package:immich_mobile/modules/album/services/album.service.dart:164)
<asynchronous suspension>
#3      SyncService._addAlbumFromServer (package:immich_mobile/shared/services/sync.service.dart:392)
<asynchronous suspension>
#4      diffSortedLists (package:immich_mobile/utils/diff.dart:30)
<asynchronous suspension>
#5      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/shared/services/sync.service.dart:267)
<asynchronous suspension>
"
2023-11-21 10:31:01.433872,LogLevel.INFO,AlbumService"",""Found 24 device albums"","""""

**Reply by joelis10** [reply]
+1 on this, seems to happen with normal albums too @alextran1502 - still an issue on 1.88.1

**Reply by alextran1502** [reply]
We might have introduced an unoptimized query while fixing the sorting on the web. Will revert it in a hot fix in a couple hours

**Reply by joelis10** [reply]
awesome, thanks for the quick reply

**Reply by alextran1502** [reply]
Cc @martabal

**Reply by alextran1502** [reply]
@Uberpainer do you have albums with a few thousands items in them?

**Reply by joelis10** [reply]
I can't speak for @Uberpainer, but I have an album with over 3k pics of my cat that takes a while to load, whereas an album with 12 photos takes about 2 seconds. Even still, 2 seconds for a small album is longer than before, where it was instant with any album.

**Reply by Uberpainer** [reply]
@alextran1502 
Album size is arround 1500 items.

**Comment by l4rm4nd** [comment]
> For those using a custom proxy, please update the routing to forward all requests to immich-server without the /api path re-write.

Do you have an example for Traefik reverse proxy? 

My old Traefik labels do not work for the new `immich-server` with the new port 3001. I assume the issue is that both the web app as well as api is handled now by the `immich-server` container. Some additional routing must be defined to make it work I guess.

These were my previous Traefik labels for the `immich-proxy` container service:

````
    labels:
      - traefik.enable=false
      - traefik.http.routers.immich.rule=Host(`immich.example.com`)
      - traefik.http.services.immich.loadbalancer.server.port=8080
      - traefik.docker.network=proxy
      # Part for local lan services only
      - traefik.http.routers.immich.middlewares=local-ipwhitelist@file
      - "com.centurylinklabs.watchtower.enable=true"
````

Moving them to `immich-server` does not work out-of-the-box:

````
services:
  immich-server:
    container_name: immich-server
    image: altran1502/immich-server:v1.88.1
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    expose:
      - 3001
    environment:
      - NODE_ENV=production
    depends_on:
      - immich-redis
      - immich-database
      - immich-typesense
    restart: unless-stopped
    labels:
      - traefik.enable=false
      - traefik.http.routers.immich.rule=Host(`immich.example.com`)
      - traefik.http.services.immich.loadbalancer.server.port=3001
      - traefik.docker.network=proxy
````

**Reply by l4rm4nd** [reply]
Update:

Ignore my post. The labels function correctly. 

I've had some old environment variable `VITE_SERVER_ENDPOINT=http://localhost:2283/api` still, which I commented out and recreated the stack with

````
docker compose pull
docker compose up -d --force-recreate --remove-orphans
````
Now it works flawlessly.

**Comment by logburn** [comment]
PSA for anyone who uses [NginxProxyManager](https://github.com/NginxProxyManager/nginx-proxy-manager) (or potentially nginx as a proxy at all) and was using custom nginx configuration similar to the one below:
```
location /api {
# Compression
gzip_static on;
gzip_min_length 1000;
gzip_comp_level 2;
proxy_buffering off;
proxy_buffer_size 16k;
proxy_busy_buffers_size 24k;
proxy_buffers 64 4k;
proxy_force_ranges on;
proxy_http_version 1.1;
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
proxy_set_header Host $host;
rewrite /api/(.*) /$1 break;
proxy_pass http://immich_server:3001;
}
```
You can simply remove this config as it is no longer needed with the simplified deployment. Thanks to the team for all the awesome work!
